Assets/Scripts/NPC/NpcConfig.cs

```diff
@@ -0,0 +1,47 @@
+using UnityEngine;
+using Framework;
+
+namespace NpcSystem
+{
+    public sealed class NpcConfig : MonoBehaviour
+    {
+        [SerializeField] private GameObject weaponPrefab;
+        [SerializeField] private Transform weaponSpawnLocation;
+        [SerializeField , Tooltip("Prefabs of the implemented animated characters")] private GameObject[] characters; 
+        [SerializeField , Tooltip("Location for the animated character to spanw(NOT world position)")] private Transform charactersSpawn;
+        [SerializeField,Tooltip("How long the player has to solve this puzzle")] private float solveTime;
+        [HideInInspector] public NpcType npcType;
```

Comment:
Ik zie de `HideInInspector` attribute. Maar wordt het ook gezet in andere classes? Zo niet, maak er een getter voor met private set.